### PR TITLE
Apply new default colours in more places

### LINKED ISSUE #5990: Scatter markers do not follow new colour cycle
See [scatter_profile](http://matplotlib.org/devdocs/examples/pylab_examples/scatter_profile.html) or [scatter_hist](http://matplotlib.org/devdocs/examples/axes_grid1/scatter_hist.html).

If I understood the code correctly, it _should_ be using `patch.facecolor`, but somewhere along the line that doesn't get applied.

### PR DESCRIPTION
**Pull request by QuLogic**
Some hardcoded defaults in `scatter`, `pie`, and `stemplot` (though now that I think about it, maybe that last one should be in `rcParams`, too.)

Also, set default colour for lines and boxplots to the new blue (and red, in some cases).

Fixes #5990.

### CODE REVIEW

**Comment by QuLogic** [comment]
Not sure why CI failed the first time, but I re-started it and it seems to be good now.

**Comment by tacaswell** [comment]
I this addresses the same issue as https://github.com/matplotlib/matplotlib/pull/5674 I believe.

**Comment by QuLogic** [comment]
Hmm, you're probably right, except that one is a bit more general and adds an additional feature for users, while this one is just straight hard-coding. Probably depends on what the consensus is for that new feature.

File: lib/matplotlib/rcsetup.py [hunk 1]
```diff
@@ -827,7 +827,7 @@ def validate_hist_bins(s):
     # line props
     'lines.linewidth':       [2.5, validate_float],  # line width in points
     'lines.linestyle':       ['-', six.text_type],             # solid line
-    'lines.color':           ['b', validate_color],  # blue
```
Comment (tacaswell) [inline lib/matplotlib/rcsetup.py]: These should change to `C0`?

**Comment by tacaswell** [comment]
I think this can be re-worked to use the new `'CN'` color specification method?

**Comment by QuLogic** [comment]
Quite likely; I think a bunch of it is already implemented, too. For the stem plots, the baseline was originally red. That's C2 for the classic colour cycle, but C3 for the new one (C2 is green).

Should I go with C2 everywhere, or just special case classic style with C3 everywhere?

**Comment by WeatherGod** [comment]
I think the red of stem plots should be treated specially, as in a separate
argument. I think when most people are thinking of specifying the color of
their stem plots, they are intending to effect only the stems themselves,
and so that should be what is impacted by the colorcycle and default color.

**Comment by QuLogic** [comment]
Yes, there is a separate argument for the stem styling; I'm only referring to the default case if nothings specified.

**Comment by QuLogic** [comment]
Rebased to fix latest v2.x build issues.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -2416,34 +2416,71 @@ def stem(self, *args, **kwargs):
...
+            linefmt = kwargs['linefmt']
+        except KeyError:
+            try:
+                linefmt = args[0]
+            except IndexError:
+                linecolor = 'C0'
+                linemarker = 'None'
+                linestyle = '-'
+            else:
+                linestyle, linemarker, linecolor = \
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: is the `\` really needed?

```python
(linestyle, linemarker,
 linecolor) =  _process_plot_format(linefmt)
```
is another way to break this up

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -2416,34 +2416,71 @@ def stem(self, *args, **kwargs):
...
+                                marker=markermarker, label="_nolegend_")
 
         if bottom is None:
             bottom = 0
 
         stemlines = []
         for thisx, thisy in zip(x, y):
-            l, = self.plot([thisx, thisx], [bottom, thisy], linefmt,
-                           label="_nolegend_")
+            l, = self.plot([thisx, thisx], [bottom, thisy],
```
Comment (tacaswell) [inline lib/matplotlib/axes/_axes.py]: might as well use `Line2D` directly here?

That might be too much to do in one pass.

File: lib/matplotlib/rcsetup.py [hunk 4]
```diff
@@ -950,7 +950,7 @@ def validate_hist_bins(s):
     'boxplot.medianprops.linewidth': [1.0, validate_float],
     'boxplot.medianprops.linestyle': ['-', six.text_type],
 
-    'boxplot.meanprops.color': ['r', validate_color],
+    'boxplot.meanprops.color': ['C3', validate_color],
```
Comment (tacaswell) [inline lib/matplotlib/rcsetup.py]: There is another PR which is touching the boxplot defaults.

**Comment by tacaswell** [comment]
This is down to just `stem` right?

All of my comments on this are non-critical.

**Pull request merged**